Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -192,4 +301,119 @@ i32 iso_days_in_month(i32 year, i32 month)
     return 28;
 }
 
+// 12.1.36 BuildISOMonthCode ( month )
+String build_iso_month_code(i32 month)
+{
+    return String::formatted("M{:02}", month);
+}
+
+// 12.1.37 ResolveISOMonth ( fields ), https://tc39.es/proposal-temporal/#sec-temporal-resolveisomonth
+double resolve_iso_month(GlobalObject& global_object, Object& fields)
+{
+    auto& vm = global_object.vm();
+
+    // 1. Let month be ? Get(fields, "month").
+    auto month = fields.get(vm.names.month);
+    if (vm.exception())
+        return {};
+
+    // 2. Let monthCode be ? Get(fields, "monthCode").
+    auto month_code = fields.get(vm.names.monthCode);
+    if (vm.exception())
+        return {};
+
+    // 3. If monthCode is undefined, then
+    if (month_code.is_undefined()) {
+        // a. If month is undefined, throw a TypeError exception.
+        if (month.is_undefined()) {
+            vm.throw_exception<TypeError>(global_object, ErrorType::TemporalMissingRequiredProperty, vm.names.month.as_string());
+            return {};
+        }
+        // b. Return month.
+        return month.as_double();
+    }
+
+    // 4. Assert: Type(monthCode) is String.
+    VERIFY(month_code.is_string());
+    auto& month_code_string = month_code.as_string().string();
+    // 5. Let monthLength be the length of monthCode.
+    auto month_length = month_code_string.length();
+    // 6. If monthLength is not 3, throw a RangeError exception.
+    if (month_length != 3) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidMonthCode);
+        return {};
+    }
+    // 7. Let numberPart be the substring of monthCode from 1.
+    auto number_part = month_code_string.substring(1);
+    // 8. Set numberPart to ! ToIntegerOrInfinity(numberPart).
+    auto number_part_integer = Value(js_string(vm, move(number_part))).to_integer_or_infinity(global_object);
+    // 9. If numberPart < 1 or numberPart > 12, throw a RangeError exception.
+    if (number_part_integer < 1 || number_part_integer > 12) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidMonthCode);
+        return {};
+    }
+    // 10. If month is not undefined, and month ≠ numberPart, then
+    if (!month.is_undefined() && month.as_double() != number_part_integer) {
+        // a. Throw a RangeError exception.
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidMonthCode);
+        return {};
+    }
+    // 11. If ! SameValueNonNumeric(monthCode, ! BuildISOMonthCode(numberPart)) is false, then
+    if (!same_value_non_numeric(month_code, js_string(vm, build_iso_month_code(number_part_integer)))) {
```

Comment:
I think we can deviate from the spec here a bit as we already know we're comparing two strings:

```suggestion
    if (month_code_string != build_iso_month_code(number_part_integer))) {
```

Reply:
done :)